force-app/main/default/classes/ALLO_AllocationsRecalculateService.cls

```diff
@@ -168,7 +168,11 @@ public inherited sharing class ALLO_AllocationsRecalculateService {
         // Allocations should not be copied to the target if the target object's amount is null
         if (amount == null) {
             return;
-        }
+        } else if (sourceAmount == 0 && convertToPercentages) {
```

Comment:
Is thee any chance that `sourceAmount` might be null (it gets populated by a call to `ALLO_AllocationsUtil.getAmountFor()`)? If so, should this method also return?